deadlines/src/main/java/com/palantir/deadlines/Deadlines.java

```diff
@@ -175,10 +178,21 @@ static String durationToHeaderValue(long durationNanos) {
                 + ((int) (durationNanos % 10000000) / 1000000);
     }
 
-    // converts a String representing seconds (or fractions thereof) to nanoseconds
+    /**
+     * Parses a String representing seconds (or fractions thereof) to nanoseconds; otherwise null if invalid.
+     */
+    @Nullable
     @VisibleForTesting
-    static long headerValueToDuration(String value) {
-        double seconds = Double.parseDouble(value);
+    static Long tryParseSecondsToNanoseconds(String value) {
+        if (Strings.isNullOrEmpty(value)) {
+            return null;
+        }
+
+        Double seconds = Doubles.tryParse(value);
```

Comment:
we can optimize this later as the Guava `Doubles.tryParse` uses a regex to validate & extract, but we only care about relatively simple (positive) decimal seconds

https://github.com/google/guava/blob/master/guava/src/com/google/common/primitives/Doubles.java#L734-L775

Reply:
https://github.com/palantir/deadlines-java/pull/19